src/main/java/teammates/common/datatransfer/attributes/StudentProfileAttributes.java

```diff
@@ -215,7 +221,7 @@ public Builder withNationality(String nationality) {
         }
 
         public Builder withGender(String gender) {
-            profileAttributes.gender = isGenderValid(gender) ? gender : "other";
+            profileAttributes.gender = isGenderValid(gender) ? Gender.valueOf(gender) : Gender.other;
```

Comment:
Instead of doing this, you should modify the parameter type of `gender` instead.